From 02f6872d91a09a9ffa35c8b43137e54d34a330e0 Mon Sep 17 00:00:00 2001
From: Tomas Glozar <tglozar@gmail.com>
Date: Mon, 22 Sep 2025 13:17:15 +0200
Subject: [PATCH 44/94] ia64: Remove ia64-specific get_order

ia64 implements its own get_order. But get_order now has a generic
implementation as well.

Move to the generic implementation. This fixes an issue where an
ia64-specific exponent routine evaluated wrongly at compile time,
causing a BUILD_BUG_ON in mm/page_frac_cache.c worked around by
13e63c2f3368374e7b30b8672edfc79c6db83079 (dropped) until now.

Signed-off-by: Tomas Glozar <tglozar@gmail.com>
---
 arch/ia64/include/asm/page.h | 13 +------------
 1 file changed, 1 insertion(+), 12 deletions(-)

diff --git a/arch/ia64/include/asm/page.h b/arch/ia64/include/asm/page.h
index 3a05cb62d729..2d37234f7f5b 100644
--- a/arch/ia64/include/asm/page.h
+++ b/arch/ia64/include/asm/page.h
@@ -116,18 +116,7 @@ typedef union ia64_va {
 extern unsigned int hpage_shift;
 #endif
 
-static __inline__ int
-get_order (unsigned long size)
-{
-	long double d = size - 1;
-	long order;
-
-	order = ia64_getf_exp(d);
-	order = order - PAGE_SHIFT - 0xffff + 1;
-	if (order < 0)
-		order = 0;
-	return order;
-}
+#include <asm-generic/getorder.h>
 
 #endif /* !__ASSEMBLY__ */
 
-- 
2.25.1

